pasqal_cloud/client.py

```diff
@@ -230,18 +206,22 @@ def get_batch_jobs(self, batch_id: str) -> list[Dict[str, Any]]:
             # the same order as they do in the GET /batches/{id} response
         )
 
+    @retry_http_error(max_retries=2)
+    def _download_results(self, results_link: str) -> JobResult:
+        response = requests.get(results_link)
+        response.raise_for_status()
+        data = response.json()
+        return JobResult(
+            raw=data.pop("raw", None), counter=data.pop("counter", None), **data
+        )
+
     def get_job_results(self, job_id: str) -> Optional[JobResult]:
```

Comment:
None of these funcs public or priv have any docstr just to account for any necessary details